### PR TITLE
fix: gate DLP program updates by delegation authority

### PR DESCRIPTION
**Pull request by thlorenz**
## Summary

Tightens program-subscription based account update forwarding so that DLP-owned account updates are only delivered when the account is actually delegated to this validator (or unconfined). Previously any DLP-owned account update that was not recognized as internal DLP state would be forwarded, which added accounts delegated to other validator into the bank without setting up a direct sub.

## Summary by CodeRabbit

* **New Features**
  * Program-sourced account updates now consult delegation records (including validator identity and confinement) before forwarding.

* **Refactor**
  * Delegation and confinement checks consolidated into a shared decision path used across discovery and subscription flows; validator identity is threaded into update clients and actors.

* **Bug Fixes**
  * RPC account fetches honor minimum-context-slot and correctly handle missing/stale/short account data.

* **Tests**
  * Added tests for delegation decisions, header parsing, fetch edge cases, and forwarding behavior.

### CODE REVIEW

File: magicblock-chainlink/src/delegation_record.rs [hunk 1]
```diff
@@ -0,0 +1,367 @@
...
+    let Some(record) = fetch_delegation_record_header(
+        rpc_client,
+        delegated_account_pubkey,
+        min_context_slot,
+    )
+    .await
+    else {
+        trace!(pubkey = %delegated_account_pubkey, slot = min_context_slot, "Dropping DLP program update without delegation record");
+        return false;
+    };
```
[the remark this answers is not in the document]
Comment (thlorenz) [reply magicblock-chainlink/src/delegation_record.rs]: Trace is fine here.

File: magicblock-chainlink/src/remote_account_provider/chain_pubsub_actor.rs [hunk 2]
```diff
@@ -718,15 +746,23 @@ impl ChainPubsubActor {
...
+                                        &validator_pubkey,
+                                        acc_pubkey,
+                                        &account.owner,
+                                        &account.data,
+                                        sub_update.slot,
+                                    )
+                                    .await
+                                } else {
+                                    false
+                                };
```
[the remark this answers is not in the document]
Comment (thlorenz) [reply magicblock-chainlink/src/remote_account_provider/chain_pubsub_actor.rs:765]: A cache is not a good idea here as we cannot EVER clone accounts that aren't delegated to us.
Also the ones that are won't see an update + in general delegated accounts are not updated very often.

We can add caching (which requires more complex machinery) to ensure correct decisions at all times once we encounter issues.

**Review by snawaz — APPROVED** [review]
LGTM.
